### ISSUE TITLE
Organize migrations by app version

### ISSUE THREAD

**Issue by thinkingmedia** [cakephp/migrations#216]
This is a (multiple allowed):
- [ ] bug
- [ ] enhancement
- [x] feature-discussion (RFC)
- CakePHP Version: 3.2.7
- Migrations plugin version: 1.5.7
- Bake plugin version (if relevant): 
- Database server (MySQL, SQLite, Postgres): MySQL
- PHP Version: PHP 7.0.5
- Platform / OS: Windows 10
### Current Behavior

Migration files are stored in a single folder with time stamps in the name for sequence.

```
  config
    + Migrations
      + 2016330301800_CreateDocuments.php
      + 2016330301810_CreateUsers.php
      + 2016330301820_CreatePosts.php
      + 2016330301830_CreateCategories.php
      + 2016330301840_UpdateUsers.php
```
### Feature Request

Migration files are stored in subfolders for each version of the app.

```
  config
    + Migrations
      + v1_0_0
        + 2016330301800_CreateDocuments.php
        + 2016330301810_CreateUsers.php
        + 2016330301820_CreatePosts.php
        + 2016330301830_CreateCategories.php
      + v1_0_1
        + 2016330301840_UpdateUsers.php
```

I don't know if the version folders would require a time stamp as well. I know that Phinx uses the time stamps as markers stored in the update table. I'm not looking to changing anything related to that. I just want to be able to organize migrations by app version.

**Comment by josegonzalez** [comment]
Why not place the version number in the migration filename?

**Comment by thinkingmedia** [comment]
@josegonzalez Too many files is one reason. After working on an app for a year or more the list will grow very long.

**Comment by HavokInspiration** [comment]
There's this PR hanging on the phinx repo : https://github.com/robmorgan/phinx/pull/716
That would solve one part of the problem (I did not look at the specifics, but it would seem). We'd be able to extend it a bit if we went to (specifying the folder name through an option for instance).

The other part would be to allow bake (whether it is for snapshot or "simple" migrations) to save the file in the desired directory. Currently the option is not available, the files are just put inside the **config/Migrations** directory.
